platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/ConfigurationAdminImpl.java

```diff
@@ -743,17 +747,50 @@ public ConfigurationStatus enableManagedServiceFactoryConfiguration(
 
     String enabledFactoryPid =
         StringUtils.removeEnd(disabledFactoryPid, ConfigurationStatus.DISABLED_EXTENSION);
-    properties.put(org.osgi.service.cm.ConfigurationAdmin.SERVICE_FACTORYPID, enabledFactoryPid);
+    Dictionary<String, Object> enabledProperties =
+        copyConfigProperties(properties, enabledFactoryPid);
+    enabledProperties.put(
+        org.osgi.service.cm.ConfigurationAdmin.SERVICE_FACTORYPID, enabledFactoryPid);
     Configuration enabledConfiguration =
         configurationAdmin.createFactoryConfiguration(enabledFactoryPid, null);
-    enabledConfiguration.update(properties);
+    enabledConfiguration.update(enabledProperties);
 
     disabledConfig.delete();
 
     return new ConfigurationStatusImpl(
         enabledFactoryPid, enabledConfiguration.getPid(), disabledFactoryPid, servicePid);
   }
 
+  private Dictionary<String, Object> copyConfigProperties(
+      Dictionary<String, Object> properties, String factoryPid) {
+    final Dictionary<String, Object> copiedProperties = new Hashtable<>();
+    ObjectClassDefinition objectClassDefinition = getObjectClassDefinition(factoryPid);
+    if (objectClassDefinition == null) {
+      LOGGER.debug(
+          "ObjectClassDefinition not found for factoryPid: {}. Unable to copy properties.",
+          factoryPid);
+      throw new IllegalStateException(
+          "ObjectClassDefinition not found for factoryPid: " + factoryPid);
+    }
+    Stream.of(objectClassDefinition)
+        .map(ocd -> ocd.getAttributeDefinitions(ObjectClassDefinition.ALL))
+        .flatMap(Arrays::stream)
+        .map(AttributeDefinition::getID)
+        .forEach(id -> copyIfDefined(id, properties, copiedProperties));
+    copyIfDefined(
```

Comment:
This will work but I am not a fan of having registry specific logic in this class. That being said if we need to do that here I would recommend just using a local constant for the property instead of making this module depend on the registry modules.